app/scripts/modules/core/src/presentation/forms/inputs/expression/ExpressionError.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { ISpelError } from './ExpressionInput';
+import { ValidationError } from 'core/validation';
+
+export interface IExpressionErrorProps {
+  spelError: ISpelError;
+}
+
+export interface IExpressionErrorState {
+  showContextModal: boolean;
+}
+
+export class ExpressionError extends React.Component<IExpressionErrorProps, IExpressionErrorState> {
+  public state: IExpressionErrorState = { showContextModal: false };
+
+  public render() {
+    const { spelError } = this.props;
+    if (!spelError || !spelError.message) {
+      return null;
+    }
+
+    const toggleModal = () => this.setState(state => ({ showContextModal: !state.showContextModal }));
+
+    return (
+      <div>
+        <Modal show={this.state.showContextModal} onHide={toggleModal}>
+          <Modal.Header>
+            <h3>{spelError.message}</h3>
+          </Modal.Header>
+
+          <Modal.Body>
+            <pre>{spelError.context}</pre>
+          </Modal.Body>
+
+          <Modal.Footer>
+            <button className="btn btn-primary" type="button" onClick={toggleModal}>
+              Close
+            </button>
+          </Modal.Footer>
+        </Modal>
+
+        <ValidationError message={spelError.message} />
+
+        <a className="clickable" onClick={toggleModal}>
```

Comment:
very nit pick (feel free to ignore) - button instead of a.

Reply:
change to `<button className="link">`?  Is your criteria that an anchor should always link to something, not just perform an action?